tests/unit/io/Cereal_unittest.cc

```diff
@@ -0,0 +1,77 @@
+#include <shogun/lib/common.h>
+#include <shogun/base/Parameter.h>
+#include <gtest/gtest.h>
+#include <../tests/unit/base/CerealObject.h>
+
+#include <shogun/lib/SGVector.h>
+#include <cereal/archives/json.hpp>
+#include <shogun/io/SerializableJsonFile.h>
+#include <fstream>
+
+using namespace shogun;
+
+TEST(Cereal, Json_SGVector_FLOAT64_save_load_pair)
+{
+	SGVector<float64_t> a(2), b;
+	a.set_const(1.14263158);
+
+	{
+		std::ofstream os("cereatltest_sgvector.cereal");
```

Comment:
We dont want that in Shogun's interface to the outside world (SWIG)
We want the save method to take a filename string instead (and an enum for the type for serialization the user wants)